M2/Macaulay2/m2/orderedmonoidrings.m2

```diff
@@ -268,7 +268,21 @@ Ring OrderedMonoid := PolynomialRing => (			  -- no memoize
 		   f=f*RM_(-minexps); -- get rid of monomial in factor if f Laurent polynomial
 		   c=RM_minexps;
 		   );
-	       (facs,exps) := rawFactor raw f;	-- example value: ((11, x+1, x-1, 2x+3), (1, 1, 1, 1)); constant term is first, if there is one
+	       isNumberField := F -> isField F and instance(baseRing F, QuotientRing) and coefficientRing baseRing F === QQ and numgens baseRing F == 1 and numgens ideal baseRing F == 1; 
```

Comment:
Would be nice if `NumberField` was a type.

Reply:
I totally agree! In fact I have some prototypical codes for a `NumberField` type. But I don't know if this should be a package or go into the core.

Reply:
Also most advanced things with number fields can be done with PARI/GP, and since the pari library is shipped with M2 (or used to be?), it would be nice to be able to use it.

Reply:
Alas, we cannot use Pari/GP in Macaulay2.  The reason has to do with memory allocation, and the desire to use other packages without patching.  Perhaps we could try it again, but I don't think we can get it to be robust (pari changes the gmp memory allocation functions, which is not compatible with use of gmp in other libraries).  But still, perhaps we can think of some way to get this functionality.

Reply:
> Yes... But to call `rawFactor` we need an extension with only one minimal polynomial. Maybe `isSimpleExtension` is a better name.

Or maybe `isSimpleNumberField`.

Reply:
Yes I ran into a lot of these problems with pari... So maybe we should just use the `gp` shell, then everything can stay in M2.